rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/serialization/NodeSourceConfigurationParser.java

```diff
@@ -114,13 +109,20 @@ private PluginDescriptor getPolicyPluginDescriptor(JSONObject jsonObject) {
     private PluginDescriptor getPluginDescriptor(JSONObject p, String pluginName) {
 
         String pluginDescription = p.get("pluginDescription").isString().stringValue();
-        JSONObject jsonSectionDescriptions = p.get("sectionDescriptions").isObject();
-        Map<Integer, String> sectionDescriptions = new HashMap<>(jsonSectionDescriptions.size());
-        for (String key : jsonSectionDescriptions.keySet()) {
-            int sectionSelector = Integer.parseInt(key);
-            String sectionDescription = jsonSectionDescriptions.get(key).isString().stringValue();
-            sectionDescriptions.put(sectionSelector, sectionDescription);
-        }
+        Map<Integer, String> sectionDescriptions = Optional.ofNullable(p.get("sectionDescriptions"))
+                                                           .map(JSONValue::isObject)
+                                                           .map(jsonSectionDescriptions -> {
+                                                               Map<Integer, String> newSections = new HashMap<>();
+                                                               for (String key : jsonSectionDescriptions.keySet()) {
+                                                                   int sectionSelector = Integer.parseInt(key);
```

Comment:
May move the declaration outside the loop

Reply:
Thanks, but I use it only inside this loop.